docs/development/commit-messages.md

```diff
@@ -0,0 +1,160 @@
+---
+layout: developer-doc
+title: Commit Messages
+category: development
+tags: [development, commit, git]
+order: 1
+---
+
+# Commit Messages
+
+Commit messages are an incredibly important part of development, as they provide
+a summary of the changes that are included in a commit. This is instrumental in
+producing a clear history for a given project.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Where these Guidelines Apply](#where-these-guidelines-apply)
+- [Formatting Your Commit Messages](#formatting-your-commit-messages)
+  - [Summary Line](#summary-line)
+  - [Detailed Explanation](#detailed-explanation)
+  - [Issue Number](#issue-number)
+  - [General Formatting](#general-formatting)
+
+<!-- /MarkdownTOC -->
+
+## Where these Guidelines Apply
+
+These guidelines apply to the following branches in the Enso repository:
+
+- The main branch (`main`).
+- Release branches (those beginning with `release/`, as specified in the
+  [release policy](../distribution/release-policy.md#release-branches)).
+
+This means that you _must_ be careful to apply these guidelines when creating
+the squash commit to merge your PRs into `main`, as that commit message must
+follow the guidelines.
+
+## Formatting Your Commit Messages
+
+Your commit messages should be formatted as follows:
+
+```plain
+[Type] A brief summary line
+
+1.  A numbered list.
+2.  That describes the changes made.
+
+Related #issue-no
+```
+
+All elements are mandatory for all commits. Please note that the blank line
+between the summary and the numbered list, and the line between the numbered
+list and the `Resolves` are crucial to prevent certain tools from getting
+confused.
+
+An example of a good commit message is as follows:
+
+```plain
+[Fix, Style] Fix a crash in the frobnicator
```

Comment:
I'd like to be able to drop spaces between the tags. It  saves a character or two, but still looks good

```
[Fix,Style] Fix a crash in the frobnicator
```